plugins/UFPWriter/UFPWriter.py

```diff
@@ -60,5 +62,48 @@ def write(self, stream, nodes, mode = MeshWriter.OutputMode.BinaryMode):
         else:
             Logger.log("d", "Thumbnail not created, cannot save it")
 
+        #Store the material.
+        application = Application.getInstance()
+        machine_manager = application.getMachineManager()
+        global_stack = machine_manager.activeMachine
+        container_registry = application.getContainerRegistry()
+
+        material_extension = "xml.fdm_material"
+        material_mime_type = "application/x-ultimaker-material-profile"
+
+        try:
+            archive.addContentType(extension=material_extension, mime_type=material_mime_type)
+        except:
+            Logger.log("w", "The material extension: %s was already added", material_extension)
+
+        added_materials = []
+        for extruder_stack in global_stack.extruders.values():
+            material = extruder_stack.material
+            material_file_name = material.getMetaData()["base_file"] + ".xml.fdm_material"
+            material_file_name = self._default_data_path + material_file_name
+
+            #Same material cannot be added
+            if material_file_name in added_materials:
+                continue
+
+            material_containers = container_registry.findContainers(id=material.getMetaDataEntry("base_file"))
+
+            if not material_containers:
+                Logger.log("e", "Cannot find material container with id: %s", material.id)
+                return False
+
+            material_container = material_containers[0]
+            try:
+                serialized_material = material_container.serialize()
+            except NotImplementedError:
+                Logger.log("e", "Unable serialize material container with id: %s", material.id)
+                return False
+
+            material_file = archive.getStream(material_file_name)
+            material_file.write(serialized_material.encode("UTF-8"))
+            archive.addRelation(virtual_path=material_file_name,relation_type="http://schemas.ultimaker.org/package/2018/relationships/xml.fdm_material")
```

Comment:
- Use `"http://schemas.ultimaker.org/package/2018/relationships/material"` for `relation_type` 
- Add `origin` to use the gcode file.